lib/commons/dom/is-focusable.js

```diff
@@ -1,5 +1,19 @@
 /* global dom */
 /* jshint maxcomplexity: 20 */
+
+/**
+ * Determines if an element is hidden in such a way that makes it unfocusable
+ * @private
+ * @param {Element} el
+ * @return {Boolean} Whether the element is unfocusable due to being hidden
+ */
+function unfocusableBecauseHidden (el) {
```

Comment:
If we keep this for performance - what about parent-child relationships? This function is a bit naive - look at isVisible for all the conditions that must be handled. Also add tests to handle all those conditions.

Reply:
Also, is it possible to do a faster implementation like https://github.com/jquery/jquery/blob/2d4f53416e5f74fa98e0c1d66b6f3c285a12f0ce/src/css/hiddenVisibleSelectors.js#L8